src/Extension/CoreExtension.php

```diff
@@ -1554,6 +1560,18 @@ function twig_get_attribute(Environment $env, Source $source, $object, $item, ar
 
             return $object->$item;
         }
+
+        if ($env->isStrictProperties()) {
+            if ($isDefinedTest) {
+                return false;
+            }
+
+            if ($ignoreStrictCheck || !$env->isStrictVariables()) {
```

Comment:
Why a check on strict variables here ?

Reply:
This avoids falling through to the method call logic below, without throwing an exception (for consistency with the chosen `strict_variables` setting).

Reply:
The `strict_variables` setting has a well-defined semantic: deciding whether undefined values are returned as `null` or throwing an exception.

Reply:
This logic aims precisely at maintaining this behavior, regardless of the value of strict_properties.